packages/core/src/http-client/http-client.ts

```diff
@@ -266,7 +266,10 @@ async function getCsrfHeaders(
     : buildCsrfHeaders(destination, {
         params: request.params,
         headers: request.headers,
-        url: request.url
+        url: request.url,
```

Comment:
why we do not pass the full request Config along? Are the parts which we do not want to use for the csrfHeader fetch?

Reply:
Here are some reasons:
- We built request with partial request config before. Using all configs might introduce issues. 
- Some keys should be replaced anyways like:
    - `method` should be `head`
    - `data` should be empty
    - `responseType` should not be some special value like `stream`/`arrayBuffer`.
    - `maxContentLength` should not be related to the `head` request (CSRF token request).